lib/active_fedora/schema_indexing_strategy.rb

```diff
@@ -0,0 +1,26 @@
+module ActiveFedora
+  ##
+  # An extension strategy to also apply solr indexes for each property.
+  # @note If how a field is indexed changes based on property, this would be a
+  #   good place to define that.
+  class SchemaIndexingStrategy
+    # @param [#index] indexer Indexer which responds to #index and returns a
+    #   block to call and manipulate the property.
+    def initialize(indexer=Indexers::NullIndexer.instance)
+      @indexer = indexer
+    end
+
+    # @param [ActiveFedora::Base] object The object to apply the property to.
+    # @param [ActiveTriples::Property, #name, #to_h] property The property to define.
+    def apply(object, property)
+      object.property property.name, property.to_h do |index|
+        indexer.index(index)
```

Comment:
Ah, this API won't work, the index object doesn't have access to the property it's indexing.